DisplayAllCharacters.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+session_start();
+
+$db = new PDO('mysql:host=db; dbname=lower_deck_charaters', 'root', 'password');
+
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+$characterDataQuery = $db->prepare(
+    "SELECT `name`, `species`, `rank`, `image` 
+        FROM `characters`;"
+);
+
+$characterDataQuery->execute();
+
+$data = $characterDataQuery->fetchAll();
```

Comment:
This is where it's time to start thinking about using some OOP - the highlighted code should now belong in a model class that handles all database interactions for characters. 

You should also look at making a Character class (entity) to represent each character.

Take a look at the very bottom of this page on the notes: https://github.com/ashleycoles/notes/blob/main/Databases.md#using-pdo-with-oop